sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/StarJoinCostBasedReorderSuite.scala

```diff
@@ -350,7 +350,7 @@ class StarJoinCostBasedReorderSuite extends PlanTest with StatsEstimationTestBas
           Some(nameToAttr("d3_c2") === nameToAttr("t1_c1")))
         .join(t5.join(t6, Inner, Some(nameToAttr("t5_c2") === nameToAttr("t6_c2"))), Inner,
           Some(nameToAttr("d2_c2") === nameToAttr("t5_c1")))
-        .select(outputsOf(d1, t3, t4, f1, d2, t5, t6, d3, t1, t2): _*)
+        .select(outputsOf(d1, t3, t4, f1, d3, d2, t5, t6, t1, t2): _*)
```

Comment:
@srowen Sorry, I haven't found a way to make the test assertions deterministic with a sort today, but I found if we move `.join(d3)` from after `.join(t6)` to before `.join(d2)`  optimization results both same as before:

```
Join Inner, (f1_fk2#1 = d2_pk#8)
:- Join Inner, (d1_pk#5 = f1_fk1#0)
:  :- Join Inner, (f1_fk3#2 = d3_pk#11)
```

Reply:
That is a good solution - assuming that the test is still valid after that reordering, and that this doesn't defeat its purpose. That I don't know yet.

Reply:
OK ~ Wait for @ioana-delaney  @cloud-fan  @gatorsmile  to help check this problem ~

Reply:
If the optimizer rule is already non-deterministic, I'm fine to update the test, and leave a TODO to make the rule deterministic later.

Reply:
@srowen @cloud-fan 

From the above test results, the result of this case is non-deterministic and using `LinkedHashMap` instead of `HashMap` has not achieved the expected effect. It may be necessary to further ensure the order of `itmes` used to generate "foundPlans".

Reply:
OK so we're saying the test is still valid with the modification, that both results are equally correct and so tweaking it to happen to come out the same in 2.12/2.13 is OK? If so I think we're good to go here.